src/utils.ts

```diff
@@ -500,3 +501,31 @@ export function getFileExtension(pathAsString: string) : string | undefined {
 	}
 	return undefined;
 }
+
+export function getAppendRegisteredSetting() : string | undefined {
+	return extensionFunctions.getContext().workspaceState.get(DIDACT_APPEND_REGISTERED_SETTING);
```

Comment:
here it is retrieving a workspacestate but what is defined in teh devfile is a "preference" which I think should be retrieved with something like `vscode.workspace.getConfiguration('didact.append').get('registry')`

Reply:
with this devfile:

```
apiVersion: 1.0.0
metadata:
  name: didact-registry-with-trace-for-settings
attributes:
  persistVolumes: 'false'
components:
  - preferences:
      didact.autoAddDefaultTutorials: false
      didact.defaultUrl: 'https://raw.githubusercontent.com/redhat-developer/vscode-didact/master/demos/markdown/simple-example.didact.md'
      che.welcome.enable: false
      didact.append.registry: '[{"name":"AppendMe","category":"AppendedCat","sourceUri":"https://raw.githubusercontent.com/redhat-developer/vscode-didact/master/examples/registry.example.didact.md"}]'
      didact.openDefaultTutorialAtStartup: true
    type: chePlugin
    reference: 'https://gist.githubusercontent.com/apupier/c1df0d7073bbfe6b3289ab068f87bbf9/raw/482c0779a11cfa26272ba6c7fe62bf6da145aff4/meta.yaml'
    alias: didact-plugin
  - type: cheEditor
    reference: 'https://che-plugin-registry-main.surge.sh/v3/plugins/eclipse/che-theia/latest/meta.yaml'
    ```
    
    based on https://github.com/apupier/vscode-didact/tree/FUSETOOLS2-1203-adding-tuts-thru-preference where I added few traces

there is this output in Didact Activity output channel:
```
Starting Didact tutorials append process 
workspacestate: [object Object] 
workspaceState.get(DIDACT_APPEND_REGISTERED_SETTING): undefined 
setting: [object Object] 
Starting Didact window with default 
```

Reply:
Interesting. Thanks for the extra details

Reply:
it appears that I can't set hidden preferences to the workspace configuration. I get this locally: "Error: Unable to write to Workspace Settings because didact.append.registry is not a registered configuration."

Adding it as a declared configuration property does solve that problem. But I'm not sure that I want this particular preference to be user-configurable if that makes any sense.

Reply:
if it is for Che environment, on components, I think that environment variable can be provided

Reply:
@apupier I don't think it's wise to have platform-specific variations unless we have no choice. It's definitely an interesting conundrum.

Reply:
That said, the environment variable seems the best option at this point. Updated the PR and will continue testing with Che